src/kernel/src/main.rs

```diff
@@ -267,24 +267,67 @@ fn main() -> ExitCode {
 
 fn exec<E: ee::ExecutionEngine>(mut ee: E, arg: EntryArg) -> ExitCode {
     // TODO: Check how the PS4 allocate the stack.
-    // TODO: We should allocate a guard page to catch stack overflow.
     info!("Allocating application stack.");
 
-    let stack = match MemoryManager::current().mmap(
+    // Assign stack size with an extra page for guard
+    let stack_size = 0x200000;
+    let guard_size = MemoryManager::current().page_size();
+    // Combine both sizes into one
+    let total_size = stack_size + guard_size;
+
+    let mut stack = match MemoryManager::current().mmap(
         0,
-        0x200000,
+        total_size,
         arg.stack_prot(),
         MappingFlags::MAP_ANON | MappingFlags::MAP_PRIVATE,
         -1,
         0,
     ) {
         Ok(v) => v,
         Err(e) => {
-            error!(e, "Allocate failed");
+            error!(e, "Stack allocation failed");
             return ExitCode::FAILURE;
         }
     };
 
+    // Set the guard page to be non-accessible
+    #[cfg(unix)]
+    {
+        use crate::memory::Protections;
+        match MemoryManager::current().mprotect(
+            unsafe { stack.as_mut_ptr().add(stack_size - guard_size) } as *mut _,
+            guard_size,
+            Protections::empty(),
+        ) {
+            Ok(_) => (),
+            Err(e) => {
+                error!(e, "Guard protection failed");
+                return ExitCode::FAILURE;
+            }
+        }
+    }
+
+    #[cfg(windows)]
+    {
+        use windows_sys::Win32::System::Memory::{VirtualProtect, PAGE_NOACCESS};
+        let mut old_protect: u32 = 0;
+
+        let locked = unsafe {
+            VirtualProtect(
```

Comment:
Why you don't use `MemoryManager::mprotect`?

Reply:
Was just going off of how you used VirtualProtect for Windows in other pieces of code as well.